### DISCUSSION TITLE
Modelling a soft pneumatic finger

### DISCUSSION THREAD

**Discussion by jgsamartin** [SofaDefrost/SoftRobots#217] [Help - Programming with SoftRobots]
Hello everyone!

I am a PhD student in the area of soft robotics. I am learning to use SOFA in order to be able to model in it the different robots I work with in the laboratory and to be able to use its different simulation capabilities and tools (SoftRobot.Inverse, SofaGym...).

Currently, I am trying to model a silicone segment. This segment is pneumatic and its movement is produced by inflating and deflating three air chambers. An example of the desired movement of the segment can be seen in the following figure (wich presents a robot of three segments):
<img width="332" alt="Robot Movement" src="https://github.com/SofaDefrost/SoftRobots/assets/74541499/2d404318-a2a9-4804-bbd3-de65069d87b1">

I have implemented in SOFA a segment with one chamber, based on the tutorial "Soft Pneumatic Gripper". As in this tutorial, I implement the air chamber with an object of type SurfacePressureConstraint and every time Ctrl + '+' is pressed, the air chamber is inflated. However, instead of swelling as I wish, the chamber acquires super strange geometries, deforming with big spikes, as you can see in the image:
<img width="172" alt="Simulation Error" src="https://github.com/SofaDefrost/SoftRobots/assets/74541499/bfc1b58f-14b8-4afb-acb3-ffb4f817b9bc">

It looks like a fixing error of one of the parameters of the simulation. However, I have tried various aspects (lowering, for example, dt, which gives even worse results) and I have not achieved any improvement.

I would be very grateful for any hint you could give me about this!

This is my code: 
[question.zip](https://github.com/SofaDefrost/SoftRobots/files/11509051/question.zip)

**Comment by EulalieCoevoet** [comment]
Hi,

It looks like there are points of your cavity's mesh that do not coincide with the surface of the silicone. This could be the problem. I'm guessing that the mapping cannot work with those points, and then it's like applying pressure on points that are free to move in any directions.

**Reply by jgsamartin** [reply]
Thank you very much!!

I will try these days to adjust correctly the two meshes to see if the problem disappears. If it does, I will mark your comment as a reply and close the topic.

Thank you very much!!!
